shell.nix

```diff
@@ -0,0 +1,49 @@
+{ pkgs ? import <nixpkgs> {} }:
+
+let
+  python = pkgs.python312.withPackages (ps: with ps; [
+    uv
+    pyqt5
+    matplotlib
+    numpy
+    opencv4
+    pytz
+    scikit-image
+    seaborn
+    scikit-learn
+    numba
+    msgpack
+    filterpy
+  ]);
+in
+pkgs.mkShell {
+  buildInputs = [
+    python
+    pkgs.qt5.full
+
+    # GUI/graphics dependencies
+    pkgs.xorg.libxcb
+    pkgs.libxkbcommon
+    pkgs.xorg.libXcursor
+    pkgs.xorg.libXrandr
+    pkgs.xorg.libXrender
+    pkgs.xorg.libXi
+    pkgs.xorg.libXext
+    pkgs.xorg.libSM
+    pkgs.tk
+    pkgs.fontconfig
+    pkgs.freetype
+    pkgs.dbus
+    pkgs.glib
+    pkgs.libGL
+    pkgs.libstdcxx5
```

Comment:
Where do they come from? From the Dockerfile? We do not need most of them, probably. Just add Python and QT and add only what is needed.